app/helpers/course/assessment/answer/programming_helper.rb

```diff
@@ -37,7 +37,7 @@ def insert_annotations(code_block_html, programming_answer_file)
     document = Nokogiri::XML.parse(code_block_html)
 
     programming_answer_file.annotations.each do |annotation|
-      insert_annotation_block(document, annotation)
+      insert_annotation_block(document, annotation) unless annotation.posts.empty?
```

Comment:
The intent is correct, but I think the the check shouldn't be at the helper level, rather at the `_annotation_topic.html.slim` level. If you ask me, it would be cleaner to let helper insert all annotations only, and not concern it with the logic within annotation.